solana-programs/anchor/audius-data/tests/follows.ts

```diff
@@ -222,8 +225,64 @@ describe("follows", function () {
       expect(decodedData.followeeHandle.seed).to.deep.equal(
         constants2.handleBytesArray
       );
-      expect(accountPubKeys[0]).to.equal(adminStorageKeypair.publicKey.toString());
-      expect(accountPubKeys[3]).to.equal(newUser1Key.publicKey.toString());
+      expect(accountPubKeys[0]).to.equal(
+        adminStorageKeypair.publicKey.toString()
+      );
+      expect(accountPubKeys[5]).to.equal(newUser1Key.publicKey.toString());
+    });
+
+    it("delegate follows user", async function () {
+      const delegate = await testCreateUserDelegate({
+        adminKeypair,
+        adminStorageKeypair: adminStorageKeypair,
+        program,
+        provider,
+        ...getURSMParams(),
+      });
+
+      // Submit a tx where user 1 follows user 2
+      const followArgs = {
+        accounts: {
+          audiusAdmin: adminStorageKeypair.publicKey,
+          authority: delegate.userAuthorityDelegateKeypair.publicKey,
+          followerUserStorage: delegate.userAccountPDA,
+          followeeUserStorage: userStorageAccount2,
+          userAuthorityDelegate: delegate.userAuthorityDelegatePDA,
+          authorityDelegationStatus: delegate.authorityDelegationStatusPDA,
+        },
+        signers: [delegate.userAuthorityDelegateKeypair],
+      };
+
+      const followTx = await program.rpc.followUser(
+        baseAuthorityAccount,
+        UserActionEnumValues.followUser,
+        { seed: delegate.handleBytesArray, bump: delegate.userBumpSeed },
```

Comment:
stuff like this is why i think we need to break apart `createUser` and `createDelegate` - here and in the below line it would be clearer as `user1.handleBytesArray` and `user2.handleBytesArray`

Reply:
sorry for the confusion, i renamed these fields to make it more clear. i thought it would be useful to create a user and delegate relationship with one call since that's used everywhere. you never create a delegate without a user. i know it gets a little more confusing because follows has multiple users. breaking apart is a little tricky but i can get to this tm if renaming is insufficient and you feel strongly about it.